### PR TITLE
delete impersonate headers

### PR DESCRIPTION
**Pull request by MatousJobanek**

### CODE REVIEW

**Review by mfrancisc — APPROVED** [review]

File: pkg/proxy/proxy_test.go [hunk 1]
```diff
@@ -425,135 +425,170 @@ func (s *TestProxySuite) TestProxy() {
...
-								req, err := http.NewRequest(tc.ProxyRequestMethod, reqPath, nil)
-								require.NoError(s.T(), err)
-								require.NotNil(s.T(), req)
-
-								for hk, hv := range tc.ProxyRequestHeaders {
-									for _, v := range hv {
-										req.Header.Add(hk, v)
-									}
+						for _, firstHeader := range rejectedHeaders {
+							for _, additionalHeader := range rejectedHeaders {
```
Comment (rajivnathan) [inline pkg/proxy/proxy_test.go:445]: I'm confused, aren't `firstHeader` and `additionalHeader` the same here? Why do we add them twice?

Comment (rajivnathan) [reply pkg/proxy/proxy_test.go]: Talked to Matous, it's intentional so that we can test combinations of the rejectedHeaders

**Review by rajivnathan — APPROVED** [review]
Thanks!

**Pull request merged**